src/DependencyInjection/CompilerPass/MiddlewarePass.php

```diff
@@ -100,34 +102,59 @@ private function registerMiddleware(ContainerBuilder $container, array $middlewa
 
             $arguments = $clientDefinition->getArguments();
 
-            $options = [];
-
             if (!empty($arguments)) {
                 $options = array_shift($arguments);
+            } else {
+                $options = [];
             }
 
             if (!isset($options['handler'])) {
-                $handlerStack = new Definition('csa_guzzle.handler_stack');
-                $handlerStack->setFactory(['GuzzleHttp\HandlerStack', 'create']);
+                $handlerStack = new Definition(HandlerStack::class);
+                $handlerStack->setFactory([HandlerStack::class, 'create']);
                 $handlerStack->setPublic(false);
-
-                $clientHandlerStackId = sprintf('csa_guzzle.handler_stack.%s', $clientId);
```

Comment:
This is useless as a unique handler is created for each guzzle client anyway. So it's not reused and the Definition is just set on the option directly and not a Reference.